libcrux-ml-kem/src/hash_functions.rs

```diff
@@ -189,20 +226,30 @@ pub(crate) mod avx2 {
         shake128_state: KeccakState,
     }
 
+    #[hax_lib::ensures(|result|
+        fstar!("$result == Spec.Utils.v_G $input"))
+    ]
     #[inline(always)]
     fn G(input: &[u8]) -> [u8; G_DIGEST_SIZE] {
         let mut digest = [0u8; G_DIGEST_SIZE];
         portable::sha512(&mut digest, input);
         digest
     }
 
+    #[hax_lib::ensures(|result|
+        fstar!("$result == Spec.Utils.v_H $input"))
+    ]
     #[inline(always)]
     fn H(input: &[u8]) -> [u8; H_DIGEST_SIZE] {
         let mut digest = [0u8; H_DIGEST_SIZE];
         portable::sha256(&mut digest, input);
         digest
     }
 
+    #[hax_lib::requires(fstar!("v $LEN < pow2 32"))]
+    #[hax_lib::ensures(|result|
+        fstar!("$result == Spec.Utils.v_PRF $LEN $input"))
```

Comment:
Shouldn't you repeat the pre-condition here?

Reply:
No, it's just an issue for `trait` and `impl`